test/ts-node-unit-tests/benchmarks/Stock/Stock-update.bench.ts

```diff
@@ -0,0 +1,69 @@
+import type { BenchmarkContext, BenchmarkResult } from '../../benchmark';
+import { performance } from 'node:perf_hooks';
+import { join } from 'node:path';
+import { generateOHLC } from '../../data-generators';
+import { setupDOM } from '../../test-utils';
+
+
+export const config = {
+    sizes: [1000, 10_000, 100_000, 1_000_000]
+};
+
+export function before(size: number) {
+
+    return {
+        fileName: `${size}-ohlc.json`,
+        func: generateOHLC.bind(undefined, size)
+  };
+}
+
+export default async function benchmarkTest(
+    {
+        size,
+        CODE_PATH,
+        data
+    }: BenchmarkContext
+): Promise<BenchmarkResult> {
+  const { win, el } = setupDOM();
+  const hc = require(join(CODE_PATH, '/highstock.src.js'))(win);
+  require(join(CODE_PATH, '/stock-tools.src.js'))(hc);
```

Comment:
This fails. 
@TorsteinHonsi jsdom does not support loading modules since v10. 
(related change -> https://www.highcharts.com/docs/extending-highcharts/extending-highcharts

`Wrapping prototype functions`)

instead of doing 

```js
  if (typeof CustomEvent === 'function') {
                window.dispatchEvent(new CustomEvent(
                    'HighchartsModuleLoaded',
                    { detail: { path: path, module: obj[path] } }
                ));
            }
```
we should do
```js
  if (typeof CustomEvent === 'function') {
                Highcharts.win.dispatchEvent(new CustomEvent(
                    'HighchartsModuleLoaded',
                    { detail: { path: path, module: obj[path] } }
                ));
            }
```

In the modules header.


PS: [this also doesn't work ](https://github.com/highcharts/highcharts-jsdom/blob/f915c5b8432cfb00982f298c141a03f6fb5c5d74/highcharts-jsdom.js#L126C8-L126C54)

Reply:
Okay, sounds good 👍